client/src/Components/QuestionsAndAnswers/BottomTabs/AddQuestion.jsx

```diff
@@ -7,53 +7,62 @@ var AddQuestion = ({chosenProduct, productId, postQuest, toggleModel}) => {
   var storage = useContext(GlobalContext);
   // var { _productId, _chosenProduct } = storage;
 
-  var [answerEntry, setAnswerEntry] = useState('');
+  var [questionEntry, setQuestionEntry] = useState('');
   var [username, setUsername] = useState('');
   var [email, setEmail] = useState('');
+  var [emailRequire, setEmailRequire] = useState(false);
+  var [require, setRequire] = useState(false);
 
   var postQuestion = (e) => {
 
-    e.preventDefault()
-    var body = {body: answerEntry, name: username, email: email, product_id: productId};
+    e.preventDefault();
+
+    setEmailRequire(false);
+    setRequire(false);
+
+    var body = {body: questionEntry, name: username, email: email, product_id: productId};
 
     if (body.body.length === 0 || body.name.length === 0 || body.email.length === 0) {
-      alert(`Please don't leave any fields blank.`)
-    } else {
-      postQuest(body)
+      setRequire(true);
+    } else if (email.indexOf('@gmail.com') > 0 || email.indexOf('@yahoo.com') > 0 || email.indexOf('@aol.com') > 0 || email.indexOf('@email.com') > 0 ) {
```

Comment:
If I am understanding this code correctly, this conditional right here is checking to make sure if the email inputted is a proper email. How did you determine this filter for correct emails? Are there specific filters given to us from the Business Requirements Document?
Wondering because I know that there are more email suffixes that exist, such as company ones and etc.

Of course, I think this filter is perfectly fine as is at our current stage of production.

Reply:
Correct, and the business docs only stated that an error should occur if the email is not in a proper format. I shall add on to anymore conditions in the future!